ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -486,6 +492,14 @@ public void shutdown() {
       log.error("Exception while waiting for pull query metrics to close", e);
     }
 
+    try {
+      pullExecutorService.shutdown();
+      pullExecutorService.awaitTermination(
+          Duration.ofSeconds(10).toMillis(), TimeUnit.MILLISECONDS);
+    } catch (final InterruptedException e) {
+      Thread.currentThread().interrupt();
```

Comment:
If someone has interrupted this thread, you can just skip awaiting termination of the executorService and log the exception.  I don't think there's any point in resetting the interrupt flag.  That's kind of the reverse of the pattern:
https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html

Reply:
This is actually discussed in good detail here: https://stackoverflow.com/questions/3976344/handling-interruptedexception-in-java

I think if there's a notion that the caller of this method may want to check the status of the flag, then there's a pattern for catch `InterruptedException` and reset the flag via `Thread.currentThread().interrupt();` as you're doing.  If there were other things further along that might block and throw `InterruptedException`, it might make sense to do what you're doing.  I'll leave it to you to assess that.  You should probably still log the exception.

Reply:
It's closed in the Autoclosable of HARouting now, basically how Andy had fixed it